File: stacks/backend/PHP/src/ChainRiceClient.php
<?php

namespace ChainRice\Bridge;

use ChainRice\Bridge\Api\AccountingApiClient;
use ChainRice\Bridge\Api\TaxApiClient;
use ChainRice\Bridge\Blockchain\BlockchainClient;
use ChainRice\Bridge\Config\ChainRiceConfig;
use ChainRice\Bridge\Model\HealthStatus;
use GuzzleHttp\Client;
use GuzzleHttp\Exception\GuzzleException;
use Monolog\Logger;
use Monolog\Handler\StreamHandler;

/**
 * Main client for connecting to ChainRice Go services.
 */
class ChainRiceClient
{
    private AccountingApiClient $accountingApi;
    private TaxApiClient $taxApi;
    private BlockchainClient $blockchainClient;
    private ChainRiceConfig $config;
    private Logger $logger;

    public function __construct(?ChainRiceConfig $config = null)
    {
        $this->config = $config ?? new ChainRiceConfig();
        $this->logger = new Logger('chainrice-bridge');
        $this->logger->pushHandler(new StreamHandler('php://stdout', Logger::INFO));

        // Initialize HTTP client
        $httpClient = new Client([
            'timeout' => $this->config->getTimeout(),
            'headers' => [
                'Content-Type' => 'application/json',
                'Accept' => 'application/json',
                'Authorization' => 'Bearer ' . $this->config->getApiKey()
            ]
        ]);

        // Initialize API clients
        $this->accountingApi = new AccountingApiClient($httpClient, $this->config);
        $this->taxApi = new TaxApiClient($httpClient, $this->config);
        $this->blockchainClient = new BlockchainClient($this->config);
    }

    /**
     * Get the accounting API client.
     */
    public function getAccountingApi(): AccountingApiClient
    {
        return $this->accountingApi;
    }

    /**
     * Get the tax API client.
     */
    public function getTaxApi(): TaxApiClient
    {
        return $this->taxApi;
    }

    /**
     * Get the blockchain client.
     */
    public function getBlockchainClient(): BlockchainClient
    {
        return $this->blockchainClient;
    }

    /**
     * Check health of all ChainRice services.
     */
    public function healthCheck(): HealthStatus
    {
        try {
            $accountingHealth = $this->accountingApi->healthCheck();
            $taxHealth = $this->taxApi->healthCheck();
            $blockchainHealth = $this->blockchainClient->healthCheck();

            $overallHealth = $accountingHealth && $taxHealth && $blockchainHealth;

            return new HealthStatus($overallHealth, $accountingHealth, $taxHealth, $blockchainHealth);
        } catch (\Exception $e) {
            $this->logger->error('Health check failed', ['error' => $e->getMessage()]);
            return new HealthStatus(false, false, false, false);
        }
    }

    /**
     * Get the configuration.
     */
    public function getConfig(): ChainRiceConfig
    {
        return $this->config;
    }
}

File: stacks/backend/JVM/src/main/java/com/chainrice/bridge/ChainRiceClient.java
package com.chainrice.bridge;

import com.chainrice.bridge.api.AccountingApiClient;
import com.chainrice.bridge.api.TaxApiClient;
import com.chainrice.bridge.blockchain.BlockchainClient;
import com.chainrice.bridge.config.ChainRiceConfig;
import com.chainrice.bridge.model.HealthStatus;
import okhttp3.OkHttpClient;
import io.grpc.Channel;
import io.grpc.Grpc;
import io.grpc.InsecureChannelCredentials;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.util.concurrent.TimeUnit;

/**
 * Main client for connecting to ChainRice Go services.
 */
public class ChainRiceClient {
    private static final Logger logger = LoggerFactory.getLogger(ChainRiceClient.class);

    private final AccountingApiClient accountingApi;
    private final TaxApiClient taxApi;
    private final BlockchainClient blockchainClient;
    private final ChainRiceConfig config;

    /**
     * Create a new ChainRice client with default configuration.
     */
    public ChainRiceClient() {
        this(new ChainRiceConfig());
    }

    /**
     * Create a new ChainRice client with custom configuration.
     */
    public ChainRiceClient(ChainRiceConfig config) {
        this.config = config;

        // Initialize HTTP client
        OkHttpClient httpClient = new OkHttpClient.Builder()
                .connectTimeout(config.getTimeout(), TimeUnit.SECONDS)
                .readTimeout(config.getTimeout(), TimeUnit.SECONDS)
                .writeTimeout(config.getTimeout(), TimeUnit.SECONDS)
                .build();

        // Initialize gRPC channel
        Channel grpcChannel = Grpc.newChannelBuilder(config.getBlockchainUrl(), InsecureChannelCredentials.create())
                .build();

        // Initialize API clients
        this.accountingApi = new AccountingApiClient(httpClient, config);
        this.taxApi = new TaxApiClient(httpClient, config);
        this.blockchainClient = new BlockchainClient(grpcChannel, config);
    }

    /**
     * Get the accounting API client.
     */
    public AccountingApiClient getAccountingApi() {
        return accountingApi;
    }

    /**
     * Get the tax API client.
     */
    public TaxApiClient getTaxApi() {
        return taxApi;
    }

    /**
     * Get the blockchain client.
     */
    public BlockchainClient getBlockchainClient() {
        return blockchainClient;
    }

    /**
     * Check health of all ChainRice services.
     */
    public HealthStatus healthCheck() {
        try {
            boolean accountingHealth = accountingApi.healthCheck();
            boolean taxHealth = taxApi.healthCheck();
            boolean blockchainHealth = blockchainClient.healthCheck();

            boolean overallHealth = accountingHealth && taxHealth && blockchainHealth;

            return new HealthStatus(overallHealth, accountingHealth, taxHealth, blockchainHealth);
        } catch (Exception e) {
            logger.error("Health check failed", e);
            return new HealthStatus(false, false, false, false);
        }
    }

    /**
     * Get the configuration.
     */
    public ChainRiceConfig getConfig() {
        return config;
    }
}
